server/commands/music.py

```diff
@@ -343,3 +350,248 @@ def ooc_cmd_random_music(client, arg):
+        return
+    
+    if args[0] == "local":
+        targets = [client]
+        musiclist = client.music_list
+    elif args[0] == "area":
+        if client not in client.area.owners and client not in client.area.area_manager.owners and not client.is_mod:
+            client.send_ooc("You should be at least cm to add a song in a musiclist!")
+            return
+        targets = client.area.clients
+        musiclist = client.area.music_list
+    else:
+        if client not in client.area.area_manager.owners and not client.is_mod:
+            client.send_ooc("You should be at least gm to add a song in a musiclist!")
+            return
+        targets = client.area.area_manager.clients
+        musiclist = client.area.area_manager.music_list
+        
+    if musiclist == []:
+        client.send_ooc("You can add a song if a musiclist is loaded in local or in area or in hub.")
+        return
+             
+    categories = []
+    for i in range(0, len(musiclist)):
+        if "category" in musiclist[i]:
+            categories.append(musiclist[i]["category"])
+        else:
+            categories.append(None)
+
+    if f"=={args[1]}==" not in categories:
+        musiclist.append({})
+        category_id = len(musiclist) - 1
+        musiclist[category_id]["category"] = f"=={args[1]}=="
+        musiclist[category_id]["songs"] = []
+    else:
+        category_id = categories.index(f"=={args[1]}==")
+
+    musiclist[category_id]["songs"].append({})
+    song_id = len(musiclist[category_id]["songs"]) - 1
+
+    musiclist[category_id]["songs"][song_id]["name"] = args[2]
+    musiclist[category_id]["songs"][song_id]["length"] = -1
+
+    if args[0] == "local":
+        path = client.music_ref
+        client.music_list = musiclist_rebuild(musiclist, path)
+    elif args[0] == "area":
+        path = client.area.music_ref
+        client.area.music_list = musiclist_rebuild(musiclist, path)
+    else:
+        path = client.area.area_manager.music_ref
+        client.area.area_manager.music_list = musiclist_rebuild(musiclist, path)
+                
+    client.server.client_manager.refresh_music(targets, True)
+    client.send_ooc(f"'{args[2]}' song has been added to '{path}' musiclist.")
+
+
+@mod_only(hub_owners=True)
+def ooc_cmd_musiclist_create(client, arg):
+    """
+    Allow you to create a musiclist!
+    <Replace> and <UseUniqueFolder> are optional arguments
+    Usage: /musiclist_create <MusiclistName> <Replace> <UseUniqueFolder>
+    """
+    if arg == "":
+        client.send_ooc(
+            "Usage: /musiclist_create <MusiclistName> <Replace> <UseUniqueFolder>"
+        )
+        return
+
+    args = shlex.split(arg)
+    args[0] = f"storage/musiclists/{derelative(args[0])}.yaml"
+
+    if os.path.isfile(args[0]):
+        raise ArgumentError(f"Musiclist {args[0]} already exists")
+
+    musiclist = [{}]
+    replace = False
+    if len(args) > 1 and args[1].lower() == "replace":
+        replace = True
```

Comment:
replace is deprecated since /toggle_replace_music and other prefs have been introduced